pytorch_lightning/utilities/cli.py

```diff
@@ -336,6 +432,25 @@ def add_arguments_to_parser(self, parser: LightningArgumentParser) -> None:
 
     def link_optimizers_and_lr_schedulers(self) -> None:
         """Creates argument links for optimizers and learning rate schedulers that specified a ``link_to``."""
+        if any(
+            True
+            for v in sys.argv
+            for optim_name in OPTIMIZER_REGISTRY
+            if re.match(fr"^--optimizer[^\S+=]*?{optim_name}?", v)
```

Comment:
Not sure this always works. If user provides "--optimizer2 Adam" then argv will be `["--optimizer2", "Adam"]` rather than `["--optimizer2 Adam"]` as given in the tests.

Reply:
Removed !